### PR TITLE
Add timestamp info to metrics in cache

### PR DESCRIPTION
**Pull request by rveznaver**
This change is [Reviewable](https://reviewable.io/reviews/criteo/biggraphite/163)

### CODE REVIEW

File: biggraphite/metadata_cache.py [hunk 1]
```diff
@@ -307,33 +309,41 @@ def _cache_get(self, metric_name):
         """Return a Metric from a the cache, None if no such metric."""
         encoded_metric_name = bg_accessor.encode_metric_name(metric_name)
         with self.__env.begin(self.__metric_to_metadata_db, write=False) as txn:
-            id_metadata = txn.get(encoded_metric_name)
```
Comment (iksaif) [inline biggraphite/metadata_cache.py]: just call that "payload" or something

File: biggraphite/metadata_cache.py [hunk 2]
```diff
@@ -307,33 +309,41 @@ def _cache_get(self, metric_name):
...
+        if not id_metadata_timestamp:
             # cache miss
             return None, False
 
         # found something in the cache
-        split = id_metadata.split(self._METRIC_SEPARATOR, 1)
-        if len(split) != 2:
+        split = id_metadata_timestamp.split(self._METRIC_SEPARATOR, 2)
+
+        if len(split) != 3:
```
Comment (iksaif) [inline biggraphite/metadata_cache.py]: what will happen with all the existing entries ?

File: biggraphite/metadata_cache.py [hunk 3]
```diff
@@ -307,33 +309,41 @@ def _cache_get(self, metric_name):
...
+        id_str, metadata_str, timestamp_str = split
         try:
             id = uuid.UUID(id_str)
         except:
             with self.__env.begin(self.__metric_to_metadata_db, write=True) as txn:
                 txn.delete(key=encoded_metric_name)
             return None, False
 
+        # Update timestamp if diff is lower than half of the TTL
+        if int(time.time()) > (int(timestamp_str) + int(self.__ttl / 2)):
```
Comment (iksaif) [inline biggraphite/metadata_cache.py]: please extract that in a separate method (the if part)

File: biggraphite/metadata_cache.py [hunk 4]
```diff
@@ -400,8 +417,14 @@ def repair(self, start_key=None, end_key=None, shard=0, nshards=1):
                     continue
 
                 metric = self._accessor.get_metric(key)
-                expected_value = self.__get_value(metric) if metric else None
-                if value != expected_value:
```
Comment (iksaif) [inline biggraphite/metadata_cache.py]: maybe factorize the code to split somewhere, and compare only what you want (uuid + json). This way to isolate the parsing part of the code

File: biggraphite/metadata_cache.py (hunk 1 above)
Comment (rveznaver) [inline biggraphite/metadata_cache.py]: done

**Pull request merged**